### PR TITLE
Add Tag[visible]

### LINKED ISSUE #9203: customize Tag component (close icon)
### What problem does this feature solve?
So I have an items that previously have been picked by user, I render them as <Tags />. also user can close this items (closable), in my case this is delete action. This delete action is sort of dangerous action, like it's gonna do some other changes related to this items. I'd like to show popconfirm before closing(deleting) this item, with some short message. I think this is also actual for select(tag mode). The end user experience is to show that is is not a simple action, and he has to be careful with this close action.

### What does the proposed API look like?
My thoughts about api is to make some 'customCloseIcon' prop - Function(onClose):ReactNode
```javascript
<Tag
    key={item.id}
    color={'#108eed'}
    closable
    onClose={() => this.deleteDangerousItem(item.id)}
    customCloseIcon={(onClose) => (
        <Popconfirm
            title="Are you sure?"
            onConfirm={onClose}
            onCancel={() => {}}
            okText="Yes"
            cancelText="No"
        >
            <Icon type="cross" />
        </Popconfirm>
    )}
>
    <span>{item.name}</span>
</Tag>
```
as far as I know it won't break api, it's still going to be animated and stuff like that

### PR DESCRIPTION
**Pull request by yesmeck**
Close #9203

### CODE REVIEW

**Comment by picodoth** [comment]
I don't see how this solves #9203...

**Comment by yesmeck** [comment]
@picodoth https://github.com/ant-design/ant-design/pull/9214#issuecomment-362200918

File: components/tag/index.tsx [hunk 1]
```diff
@@ -33,21 +35,37 @@ export default class Tag extends React.Component<TagProps, TagState> {
     closable: false,
   };
 
-  constructor(props: TagProps) {
-    super(props);
+  static getDerivedStateFromProps(nextProps: TagProps) {
+    return  ('visible' in nextProps) ? { visible: nextProps.visible } : null;
```
Comment (picodoth) [inline components/tag/index.tsx]: 返回null的时候是覆盖state为null 还是不修改state?

Comment (yesmeck) [reply components/tag/index.tsx]: 不修改

**Review by picodoth — APPROVED** [review]
+1 after fixing ci

**Pull request merged**